pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -314,25 +298,18 @@ func getPoolByListenerID(client *gophercloud.ServiceClient, loadbalancerID strin
 			for _, l := range p.Listeners {
 				if l.ID == listenerID {
 					listenerPools = append(listenerPools, p)
+					break
 				}
 			}
 		}
-		if len(listenerPools) > 1 {
-			return false, ErrMultipleResults
-		}
 		return true, nil
 	})
 	if err != nil {
-		if isNotFound(err) {
-			return nil, ErrNotFound
-		}
 		return nil, err
 	}
 
 	if len(listenerPools) == 0 {
 		return nil, ErrNotFound
-	} else if len(listenerPools) > 1 {
-		return nil, ErrMultipleResults
```

Comment:
I agree the check should never fire, but I'm nervous about trusting the OpenStack API to always do what we expect :)

On the other hand, I agree these additional checks add visual noise - so I only have a weak preference for keeping them.